### PR TITLE
weather-task

### PR DESCRIPTION
**Pull request by mansoorahmed786**

### CODE REVIEW

File: New folder/README.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+## --range
+It will take the range of date in whcih we have to analyze data
+
+### Validations
+- It will check that date is in right format i.e 'YYYY-MM-DD to YYYY-MM-DD'
+- It will check that given range is present in file or not
+- It will check that start data ca't be greater than end date'
+
+```
+python task.py --file weather_1.csv  --range "2017-01-03 to 2016-01-03" --format values.csv
```
Comment (talhamasood0000) [inline New folder/README.md]: format => output
--input-file

File: New folder/README.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+## --state
+ It will return maximum occuring state in the given file
+
+```
+python task.py --state --file weather_1.csv
+```
+Output:
+
+Alaska
```
Comment (talhamasood0000) [inline New folder/README.md]: --max column_name
log(Maximum Occuring State in the file is:)

File: New folder/README.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+ This command will read the will __weather_1.csv__ and print the most occuring state from the file
+
+## --analyzestate
+ It will print Average (Max and Min) Temperature, Average Wind Direction, Average wind speed for given date range from a given file
+
+```
+python task.py --analyzestate --file weather_1.csv --range '2016-01-03 to 2016-01-03'
```
Comment (talhamasood0000) [inline New folder/README.md]: --average-data

File: New folder/README.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+```
+Output:
+
+ INFO:root:Imported data from weather_1.csv
+
+ INFO:root:Data is validating
+
+ State is ['Key West, FL']
+
+ Year is ['2016']
```
Comment (talhamasood0000) [inline New folder/README.md:105]: 2016

File: New folder/README.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+ Month is ['1']
+
+ Day is ['2016-01-03']
+
+ This command will read the file __weather_1.csv__ and print the state, Year, Month and Day which have maximum temprature from range '2016-01-03 to 2016-01-03'
+
+## --location
+ It will print station, station_code , station_location and station_state of maximum average temprate from a given range of date  from a whole file
```
Comment (talhamasood0000) [inline New folder/README.md:114]: --get-location

File: New folder/README.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+ ## --explain
+
+ ```
+  python task.py --explain weather_1.csv
+ ```
+
+ Output:
+
+Minimum date is 2016-01-03 00:00:00
```
Comment (talhamasood0000) [inline New folder/README.md]: Minimum date is 2016-01-03

File: New folder/args_parser.py [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+import os
+from datetime import datetime
+import pandas as pd
+import logging
+import csv
+
+class CustomArgumentParser(argparse.ArgumentParser):
+    def __init__(self):
+        super(CustomArgumentParser, self).__init__()
+        self.add_argument('--file', type=str, help='Enter File name')
```
Comment (talhamasood0000) [inline New folder/args_parser.py]: example.csv

File: New folder/args_parser.py [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+        self.add_argument('--range', type=str, help='Enter range of date')
```
Comment (talhamasood0000) [inline New folder/args_parser.py]: YYYY-MM-DD to YYYY-MM-DD

File: New folder/args_parser.py [hunk 9]
```diff
@@ -0,0 +1,317 @@
...
+        start_date = date_range[0]
+        end_date = date_range[1]
+
+        if datetime.strptime(start_date, '%Y-%m-%d') > datetime.strptime(end_date, '%Y-%m-%d'):
+            logging.info("Start date cannot be greater than end date.")
+            return False
+
+        return True
+
+    def clean_data_file(file_path):
```
Comment (talhamasood0000) [inline New folder/args_parser.py]: remove

File: New folder/args_parser.py [hunk 10]
```diff
@@ -0,0 +1,317 @@
...
+            print(f"Year is {result[1]}")
+            print(f"Month is {result[2]}")
+            print(f"Day is {result[3]}")
+
+        if args.location and args.range:
+            location.location_range(args.location,args.range)
+        elif args.location:
+            location.location(args.location)
+
+    def validate_commands(self, command):
```
Comment (talhamasood0000) [inline New folder/args_parser.py]: ?????

File: New folder/args_parser.py [hunk 11]
```diff
@@ -0,0 +1,317 @@
...
+                  Minimum wind direction is {min_wind_direction}
+                  Minimum wind speed is {min_wind_speed} 
+                  Max_temprate is {max_temp}''')
+        print(f'''Maximum max temprate is {max_max_temp}
+                  Maximum min temprature is {max_min_temp} 
+                  Maximum wind direction is {max_wind_direction} 
+                  Max wind speed is {max_wind_speed}''')
+
+
+class import_data:
```
Comment (talhamasood0000) [inline New folder/args_parser.py]: ???

File: New folder/args_parser.py [hunk 12]
```diff
@@ -0,0 +1,317 @@
...
+            wind_speed = ([entry.get('wind_speed') for entry in filtered_data])
+            average_temperature = sum(int(num) for num in list)/len(list)
+            average_max_temperature = sum(int(num) for num in max_temp)/len(max_temp)
+            average_min_temperature = sum(int(num) for num in min_temp)/len(min_temp)
+            average_wind_direction = sum(int(num) for num in wind_direction)/len(wind_direction)
+            average_wind_speed = sum(float(num) for num in wind_speed)/len(wind_speed)
+            min_temperature = min(int(num) for num in list)
+            max_temperature = max(int(num) for num in list)
+            # state = ([entry.get('station_location','station_state') for entry in filtered_data if int(entry.get('avg_temp'))==max_temperature])
+            # month = ([entry.get('date_month') for entry in filtered_data if max_temperature==int(entry.get('avg_temp'))])
```
Comment (talhamasood0000) [inline New folder/args_parser.py]: remove

File: New folder/README.md (hunk 6 above)
Comment (mansoorahmed786) [reply New folder/README.md]: Done

File: New folder/args_parser.py (hunk 7 above)
Comment (mansoorahmed786) [reply New folder/args_parser.py]: Done

File: New folder/args_parser.py (hunk 8 above)
Comment (mansoorahmed786) [reply New folder/args_parser.py]: Done

File: New folder/args_parser.py (hunk 9 above)
Comment (mansoorahmed786) [reply New folder/args_parser.py]: Done

File: New folder/args_parser.py (hunk 10 above)
Comment (mansoorahmed786) [reply New folder/args_parser.py]: Done

File: New folder/args_parser.py (hunk 12 above)
Comment (mansoorahmed786) [reply New folder/args_parser.py]: Done

File: New folder/README.md (hunk 5 above)
Comment (mansoorahmed786) [reply New folder/README.md:114]: Done

File: New folder/README.md (hunk 3 above)
Comment (mansoorahmed786) [reply New folder/README.md]: Done

File: New folder/README.md (hunk 2 above)
Comment (mansoorahmed786) [reply New folder/README.md]: Done

File: New folder/README.md (hunk 1 above)
Comment (mansoorahmed786) [reply New folder/README.md]: Done

**Pull request merged**